File: solutions/python/taskmanagementsystem/comment.py
import uuid
from datetime import datetime
from user import User

class Comment:
    def __init__(self, content: str, author: User):
        self._id = str(uuid.uuid4())
        self._content = content
        self._author = author
        self._timestamp = datetime.now()
    
    @property
    def author(self) -> User:
        return self._author

File: solutions/typescript/src/TaskManagement/Comment.ts
import User from "./User";
import { v4 as uuidv4 } from "uuid";

export default class Comment {
    private id: string;
    private content: string;
    private author: User;
    private timestamp: Date;

    constructor(content: string, author: User) {
        this.id = uuidv4();
        this.content = content;
        this.author = author;
        this.timestamp = new Date();
    }

    getAuthor() {
        return this.author;
    }
}
